### DISCUSSION TITLE
How to read a static item from a dfs2 file

### DISCUSSION THREAD

**Discussion by gaspermeister** [DHI/mikeio#676] [Q&A] [answered]
I am trying to read data from a DFS2 file (M21 result files) and do some calculations using both dynamic and static data. 
I have a lot of M21 result files that include "Water Depth" and I would like to use MIKEIO to calculate "Surface Elevation" using the bathymetry data, which is stored in theese files as a static item. I know how to read dynamic items (with ds=mikeio.read(path) and da=ds[['Water Depth']]) and if needed transform it to a numpy array (to_numpy) for different calculations. If I understand correcty, the bathymetry data can be accessed somehow through ds.geometry, but I can't fid a way to read it as an array and use it in calculations. Help would be much appreciated.

**Comment by ecomodeller** [comment, marked as answer]
The geometry is only related to the grid, it has no information about bathymetry.

You can read static item data using the lower level library mikecore, which MIKE IO uses (so you already have it installed).

Here is an example of reading a static item from a dfsu file:
https://github.com/DHI/mikecore-python/blob/master/tests/test_dfs_static_item.py

I suppose you should be able to do the same with a Dfs2 file, you only need to reshape the data to a proper 2d array.

**Reply by gaspermeister** [reply]
Thank you! I managed to read the bathymetry static data and transfrom it from 1D to 2D array based on mesh shape (which is quite straightforward for a rectangular mesh).

**Comment by ecomodeller** [comment]
#682
